shared/base/src/commonMain/kotlin/kmp/shared/analytics/domain/model/ToastAnalytics.kt

```diff
@@ -0,0 +1,17 @@
+package kmp.shared.analytics.domain.model
+
+/**
+ * Sample analytics object showing how to define analytics events
+ * and organize them in a structured way.
+ */
+object ToastAnalytics {
+
+    object Event {
+        fun presented(type: ViewType) = AnalyticsEvent(AnalyticName.Event.ToastPresented, mapOf(AnalyticName.Param.PresentedFrom to type.value))
```

Comment:
Hmm.. Following up to what I said above.. I don't find this separation ideal, because you still have to define feature-specific events and parameters in the shared `AnalyticsEvent.kt` file, but you also have to define ways how to initialize these events in these separate files.. But you'll always have to add stuff to both, which indicates some case of tight coupling.

What about having the `AnalyticsEvent` superclass with generic `String` event name and `Map<String, String>` parameters, and then we'll just define subclasses in these separate files? We might loose the advantage of knowing all defined enum cases, but I don't see any big benefit in that.